<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>759</width>
    <height>679</height>
   </rect>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::DefaultContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>TinyCom</string>
  </property>
  <property name="windowIcon">
   <iconset resource="res/tinycom.qrc">
    <normaloff>:/icons/32x32/tinycom.png</normaloff>:/icons/32x32/tinycom.png</iconset>
  </property>
  <property name="iconSize">
   <size>
    <width>32</width>
    <height>32</height>
   </size>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <property name="horizontalSpacing">
       <number>6</number>
      </property>
      <property name="verticalSpacing">
       <number>2</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <item row="1" column="1">
       <widget class="QCheckBox" name="echo_input">
        <property name="toolTip">
         <string>Put any input in the output window automatically.</string>
        </property>
        <property name="text">
         <string>Local Echo</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="btn_open">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>&amp;Open Device</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="remove_escape">
        <property name="toolTip">
         <string>Remove ANSI escape sequences in the output and log.</string>
        </property>
        <property name="text">
         <string>Remove ANSI Escape Sequences</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="serial_str">
        <property name="text">
         <string>Not connected</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QFrame" name="frame">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="lineWidth">
        <number>0</number>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QPlainTextEdit" name="log">
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Output log.</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item>
           <widget class="QPushButton" name="btn_clear">
            <property name="toolTip">
             <string>Clear output log window.</string>
            </property>
            <property name="text">
             <string>C&amp;lear</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="output_hex">
            <property name="text">
             <string>Hex Output</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QCheckBox" name="enable_log">
              <property name="toolTip">
               <string>Enable logging to file.</string>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Log To:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="log_file">
              <property name="toolTip">
               <string>Log file path.</string>
              </property>
              <property name="text">
               <string>tinycom.log</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btn_open_log">
              <property name="toolTip">
               <string>Select log file.</string>
              </property>
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="frame_2">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="lineWidth">
        <number>0</number>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QListWidget" name="history">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Input history</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetMaximumSize</enum>
      </property>
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Input:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="input">
        <property name="toolTip">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="line_end">
        <property name="toolTip">
         <string>Select line ending type to append.</string>
        </property>
        <item>
         <property name="text">
          <string>LF line end</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>CR line end</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>CR/LF line end</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>LF/CR line end</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>No line end</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Hex</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_send">
        <property name="toolTip">
         <string>Send the input.</string>
        </property>
        <property name="text">
         <string>&amp;Send</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>759</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuAbout"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="res/tinycom.qrc"/>
 </resources>
 <connections/>
</ui>
